src/pperm.cc

```diff
@@ -3657,67 +3657,16 @@ static Obj FuncOnPosIntSetsPartialPerm(Obj self, Obj set, Obj f)
     RequireSmallList("OnPosIntSetsPartialPerm", set);
     RequirePartialPerm("OnPosIntSetsPartialPerm", f);
 
-    UInt2 *     ptf2;
-    UInt4 *     ptf4;
-    UInt        deg;
-    const Obj * ptset;
-    Obj *       ptres, tmp, res;
-    UInt        i, k, reslen;
+    const UInt len = LEN_LIST(set);
 
-    if (LEN_LIST(set) == 0)
+    if (len == 0)
         return set;
```

Comment:
Note that this is another dangerous compared to `OnSets`: if `set` is an empty *mutable* set, then `OnSets` returns a new (non-identical) empty list, which helps avoid all kinds of nasty bugs. I didn't touch it here for now as I wasn't sure if this was intentional or not...